packages/fork-choice/src/forkChoice/forkChoice.ts

```diff
@@ -146,6 +146,13 @@ export class ForkChoice implements IForkChoice {
     }
   }
 
+  /**
+   * Successful call if no critical errors, else throws
+   */
+  checkSaneForkChoice(): void {
+    this.protoArray.findHead(this.fcStore.justified.checkpoint.rootHex, this.fcStore.currentSlot);
```

Comment:
Why not just return the stored error?

```ts
isProtoArrayInIrrecoverableError(): Error | null
```

Reply:
aren't the kind of errors thrown in proto-array's `findHead` all irrecoverable? lvh Error is just one kind
If they are, then i can do a small refactor of findHead and break it into two parts, one to check and get the error (which can be used as you suggested), and original findHead will use the new error inspect fn and throw if there is one.

Reply:
Calling findHead over and over just to check if there's an error feels wrong

Reply:
approach a:
```ts
if (forkChoice.irrecoverableError) {
  processShutdownCallback(forkChoice.irrecoverableError)
}
```
approach b
```
try {
  forkChoice.protoArray.findHead(this.fcStore.justified.checkpoint.rootHex, this.fcStore.currentSlot);
} catch (e) {
  processShutdownCallback(forkChoice.irrecoverableError)
}
```

approach a looks much more sane to me.